### PR TITLE
Port SwiftLangSupport::resolvePathSymlinks to Windows

### PR DESCRIPTION
**Pull request by hughbe**
@compnerd could you sign off on this change, it's less trivial than the others!

### CODE REVIEW

**Comment by hughbe** [comment]
@swift-ci please smoke test

File: tools/SourceKit/lib/SwiftLang/SwiftLangSupport.cpp [hunk 1]
```diff
@@ -756,10 +762,36 @@ bool SwiftLangSupport::printAccessorUSR(const AbstractStorageDecl *D,
...
+#else
+  char full_path[MAX_PATH];
+
+  HANDLE fileHandle = CreateFileA(
+      InputPath.c_str() /*lpFileName*/, GENERIC_READ /*dwDesiredAccess*/,
+      0 /*dwShareMode*/, nullptr /*lpSecurityAttributes */,
+      OPEN_EXISTING /*dwCreationDisposition*/,
+      FILE_FLAG_BACKUP_SEMANTICS |
+          FILE_FLAG_OPEN_REPARSE_POINT /*dwFlagsAndAttributes*/,
+      nullptr /*hTemplateFile*/);
```
Comment (compnerd) [inline tools/SourceKit/lib/SwiftLang/SwiftLangSupport.cpp]: We generally don't do naming on the parameters unless they are trailing booleans.

File: tools/SourceKit/lib/SwiftLang/SwiftLangSupport.cpp [hunk 2]
```diff
@@ -756,10 +762,36 @@ bool SwiftLangSupport::printAccessorUSR(const AbstractStorageDecl *D,
...
+
+  if (fileHandle == INVALID_HANDLE_VALUE) {
+    return InputPath;
+  }
```
Comment (compnerd) [inline tools/SourceKit/lib/SwiftLang/SwiftLangSupport.cpp]: Unnecessary braces.

File: tools/SourceKit/lib/SwiftLang/SwiftLangSupport.cpp [hunk 3]
```diff
@@ -756,10 +762,36 @@ bool SwiftLangSupport::printAccessorUSR(const AbstractStorageDecl *D,
...
+
+  DWORD success = GetFinalPathNameByHandleA(
+      fileHandle /*hFile*/, full_path /*lpszFilePath*/,
+      sizeof(full_path) /*cchFilePath*/, FILE_NAME_NORMALIZED /*dwFlags*/);
+  if (success) {
```
Comment (compnerd) [inline tools/SourceKit/lib/SwiftLang/SwiftLangSupport.cpp]: Why not use a ternary?

    return (success ? full_path : InputPath);

File: tools/SourceKit/lib/SwiftLang/SwiftLangSupport.cpp [hunk 4]
```diff
@@ -756,10 +762,36 @@ bool SwiftLangSupport::printAccessorUSR(const AbstractStorageDecl *D,
...
+          FILE_FLAG_OPEN_REPARSE_POINT /*dwFlagsAndAttributes*/,
+      nullptr /*hTemplateFile*/);
+
+  if (fileHandle == INVALID_HANDLE_VALUE) {
+    return InputPath;
+  }
+
+  DWORD success = GetFinalPathNameByHandleA(
+      fileHandle /*hFile*/, full_path /*lpszFilePath*/,
+      sizeof(full_path) /*cchFilePath*/, FILE_NAME_NORMALIZED /*dwFlags*/);
```
Comment (compnerd) [inline tools/SourceKit/lib/SwiftLang/SwiftLangSupport.cpp]: Drop the parameter names.

**Review by compnerd — CHANGES_REQUESTED** [review]

**Comment by hughbe** [comment]
@swift-ci please smoke test and merge

**Comment by hughbe** [comment]
@swift-ci please smoke test and merge

**Comment by hughbe** [comment]
Cheers, I've addressed your comments

**Comment by hughbe** [comment]
@swift-ci please smoke test

**Pull request merged**